src/app/authentication/screens/register-screen.component.tsx

```diff
@@ -1,63 +1,106 @@
+    const dispatch = useDispatch();
     const themeKey = useSelector(getTheme);
     const theme = useMemo(() => themesCollection[themeKey] as DefaultTheme, [themeKey]);
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [repeatPassword, setRepeatPassword] = useState('');
+    const [name, setName] = useState('');
+    const [validation, setValidation] = useState(true);
+
+    const handleRegister = () => {
+        if (password === repeatPassword) {
+            dispatch(REGISTER.TRIGGER({ email: email, username: name, password: password }));
+            console.log(users);
+        }
+    };
 
     return (
-        <Container>
-            {theme.colors.main === 'black' ? (
-                <Logo source={require('../../../assets/dark-logo.jpg')} />
-            ) : (
-                <Logo source={require('../../../assets/light-logo.jpg')} />
-            )}
-            <Title>Splash</Title>
-            <RegisterInputArea>
-                <RegisterInputText>{I18n.t(en.login)}</RegisterInputText>
-                <Input />
-                <PasswordText>{I18n.t(en.hello)}</PasswordText>
-                <Input />
-                <PasswordText>Password</PasswordText>
-                <Input secureTextEntry={true} />
-                <PasswordText>Password</PasswordText>
-                <Input secureTextEntry={true} />
-                <RegisterButton activeOpacity={0.5}>
-                    <LoginText>Sign in</LoginText>
-                </RegisterButton>
-            </RegisterInputArea>
-        </Container>
+        <AvoidingContainer>
+            <KeyboardAvoidingView
+                contentContainerStyle={{ flex: 1 }}
+                style={{ flex: 1 }}
+                behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
+            >
+                {themeKey === ThemesEnum.DARK && ThemesEnum.JAPANESE ? (
+                    <BackgroundImage source={require('../../../assets/background.jpg')} />
+                ) : (
+                    <BackgroundImage source={require('../../../assets/light-background.jpg')} />
+                )}
+                <LogoWrapper>
+                    <BackButtonContainer onPress={() => props.navigation.goBack()}>
+                        <Icon name={'chevron-back'} color={theme.colors.secondary} size={36} />
+                    </BackButtonContainer>
+                    <LogoContainer>
+                        <Title>Splash</Title>
+                    </LogoContainer>
+                </LogoWrapper>
+                <ScrollView showsVerticalScrollIndicator={false}>
+                    <RegisterInputArea>
+                        <EmailText>{I18n.t('Name')}</EmailText>
+                        <Input onChangeText={(val) => setName(val)} />
+                        <InputText>{I18n.t('Email')}</InputText>
+                        <Input
+                            //eslint-disable-next-line react-native/no-color-literals
```

Comment:
Move color to theme :p

Reply:
BTW you don't need to provide style prop for style component. You should add custom props when you creating it:

```
const Bruh = styled.View<{isValid: boolean}>`
    background-color: ${isValid ? ... : ...}
`;

...

<Bruh isValid={isValidState}/>
```